tests/tensorflow/quantization/test_unified_scales.py

```diff
@@ -0,0 +1,106 @@
+ See the License for the specific language governing permissions and
+ limitations under the License.
+"""
+
+import itertools
+import pytest
+
+import tensorflow as tf
+from tensorflow.python.keras import layers
+
+from nncf.common.hardware.config import HWConfigType
+from nncf.tensorflow.quantization.utils import collect_fake_quantize_layers
+from tests.tensorflow.helpers import create_compressed_model_and_algo_for_test
+from tests.tensorflow.quantization.utils import get_basic_quantization_config
+
+
+def get_single_concat_test_model(input_shapes):
+    inputs = []
+    for i, input_shape in enumerate(input_shapes):
+        inputs.append(tf.keras.Input(shape=input_shape[1:], name='input_{}'.format(i + 1)))
+    # pylint: disable=unbalanced-tuple-unpacking
+    input_1, input_2 = inputs
+
+    x_1 = layers.Multiply()([input_1, input_1])
+    x_2 = layers.Multiply()([input_2, input_2])
+
+    outputs = layers.Concatenate(1)([x_1, x_2])
+    outputs = layers.Conv2D(filters=1, kernel_size=1)(outputs)
+    return tf.keras.Model(inputs=inputs, outputs=outputs)
+
+
+def get_double_concat_test_model(input_shapes):
+    inputs = []
+    for i, input_shape in enumerate(input_shapes):
+        inputs.append(tf.keras.Input(shape=input_shape[1:], name='input_{}'.format(i + 1)))
+    # pylint: disable=unbalanced-tuple-unpacking
+    input_1, input_2 = inputs
+
+    x_1 = input_1 * input_1
+    x_2 = input_2 * input_2
+
+    cat_1 = layers.Concatenate(1)([x_1, x_2])
+    cat_2 = layers.Concatenate(1)([x_1, cat_1])
+    outputs = layers.Conv2D(filters=3, kernel_size=3, strides=2, padding='same')(cat_2)
+    return tf.keras.Model(inputs=inputs, outputs=outputs)
+
+
+def get_unet_like_test_model(input_shapes):
+    inputs = []
+    for i, input_shape in enumerate(input_shapes):
+        inputs.append(tf.keras.Input(shape=input_shape[1:], name='input_{}'.format(i + 1)))
+    # pylint: disable=unbalanced-tuple-unpacking
+    input_1, _ = inputs
+
+    conv_1 = layers.Conv2D(filters=8, kernel_size=1)(input_1)
+    conv_2 = layers.Conv2D(filters=16, kernel_size=1)(conv_1)
+    conv_3 = layers.Conv2D(filters=32, kernel_size=1)(conv_2)
+    conv_t_3 = layers.Conv2DTranspose(filters=16, kernel_size=1)(conv_3)
+
+    cat_1 = layers.Concatenate(0)([conv_t_3, conv_2])
+    conv_t_2 = layers.Conv2DTranspose(filters=8, kernel_size=1)(cat_1)
+
+    cat_2 = layers.Concatenate(0)([conv_t_2, conv_1])
+    outputs = layers.Conv2DTranspose(filters=4, kernel_size=1)(cat_2)
+    return tf.keras.Model(inputs=inputs, outputs=outputs)
+
+
+CAT_UNIFIED_SCALE_TEST_STRUCTS = [(get_single_concat_test_model, 3, 4),
+                                  (get_double_concat_test_model, 3, 4),
+                                  (get_unet_like_test_model, 4, 6)
+                                  ]
+
+
+def get_total_quantizations(model: tf.keras.Model) -> int:
+    fq_layers = [layer for layer in model.get_config()['layers'] if layer['class_name'] == 'FakeQuantize']
```

Comment:
I suggest looking at `collect_fake_quantize_layers()` method from the `nncf.tensorflow.quantization.utils` module.

Reply:
`collect_fake_quantize_layers` is used to calculate non_weight_quantizers